recipes/tiny-aes-c/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.32.0"
+
+class TinyAesCConan(ConanFile):
+    name = "tiny-aes-c"
+    license = "Unlicense"
+    homepage = "https://github.com/kokke/tiny-AES-c"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Small portable AES128/192/256 in C"
+    topics = ("encryption", "crypto", "AES")
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+
+    _options_dict = {
+        # enable AES128
+        "aes128": [True, False],
+        # enable AES192
+        "aes192": [True, False],
+        # enable AES256
+        "aes256": [True, False],
+        # enable AES encryption in CBC-mode of operation
+        "cbc": [True, False],
+        # enable the basic ECB 16-byte block algorithm
+        "ecb": [True, False],
+        # enable encryption in counter-mode
+        "ctr": [True, False],
+    }
```

Comment:
Actually upstream conan file (or header file) is broken in its logic I think due to what happen in the header file https://github.com/kokke/tiny-AES-c/blob/e72b6eff0884673997d0ca6385169bbd9b31936d/aes.h#L13-L41

What I understand:
- CBC, ECB, CTR modes:
  - if definition is undefined or defined to 1, mode is enabled. If definition is defined to 0, mode is disabled.
  - multiple modes can be enabled
  - the default is all modes enabled
- AES128, AES192 and AES256:
  - AES256 enabled if AES256 defined (to 1), AES192 enabled is AES196 defined (to 1), AES128 otherwise (the default one).
  - exclusive options, there should be only one conan option actually instead of these 3 options.